packages/expo-ui/ios/ListItem.swift

```diff
@@ -0,0 +1,65 @@
+// Copyright 2025-present 650 Industries. All rights reserved.
+
+import ExpoModulesCore
+import SwiftUI
+
+internal class SwipeAction: Record, Identifiable {
+  required init() { }
+  @Field var id: String
+  @Field var props: ButtonProps
+}
+
+class ListItemViewProps: ExpoSwiftUI.ViewProps {
+  @Field var leadingActions: [SwipeAction]
+  @Field var allowsFullSwipeLeading = true
+
+  @Field var trailingActions: [SwipeAction]
+  @Field var allowsFullSwipeTrailing = true
+  @Field var hideTrailingActionsInEditMode = true
+
+  var onActionPressed = EventDispatcher()
+}
+
+struct ActionButtons: View {
+  let fromArray: [SwipeAction]?
+  let handleButtonPress: (_ id: String) -> Void
+
+  var body: some View {
+    ForEach(fromArray ?? []) { elem in
+      elem.props.onButtonPressed.onEventSent = { _ in
+        handleButtonPress(elem.id)
+      }
+
+      return ExpoUI.Button().environmentObject(elem.props)
+    }
+  }
+}
+
+struct ListItemView: ExpoSwiftUI.View {
+  @Environment(\.editMode) private var editMode
+  @EnvironmentObject var props: ListItemViewProps
+
+  var body: some View {
+    UnwrappedChildren { child, isHostingView in
```

Comment:
I opted for this view, because it worked the best when rendering both `@expo/ui` components and `react-native` components.

When using plain `Children()`, the components from `react-native` were rendering misaligned.

Another thing (I believe known upstream in react-native repo) is the `react-native` view resizing is throttled and the size can flicker when resizing the parent view. This is noticeable when toggling edit mode